### ISSUE TITLE
[3.x]:  Invalid field layout element config: Invalid field UID on assets

### ISSUE THREAD

**Issue by joelstrom** [craftcms/cms#12384]
### Description
We have a project with multiple environments, and on one of these environments we are all of a sudden unable to use asset fields or access the assets library. Trying to click an asset field to update a file or trying to access the library throws the following error in the log:

2022-11-28 11:17:29 [-][1][-][warning][craft\models\FieldLayoutTab::init] Invalid field layout element config: Invalid field UID: 5c1a78f7-f556-4439-8c70-6b756d59f99f
2022-11-28 11:17:29 [-][1][-][error][yii\base\InvalidArgumentException] yii\base\InvalidArgumentException: Invalid field UID: 5c1a78f7-f556-4439-8c70-6b756d59f99f in /var/www/alandsbanken/releases/20221125161634Z/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php:109

The internal error message appears in the dashboard when this happens, and in the console we can see that the following action gives 500: admin/actions/element-indexes/get-elements

The assets in the library view do not appear at all as trying to interact with it gives the error right away, and therefore it's impossible to manage assets on the environment in question.

Here are some things that we have done when trying to solve the issue:

- Run the Craft utility to fix duplicated element UIDs
- Searched the database table "craft_entries" for the UID in question, no result
- Rebuild & sync project config
- Clear all caches
- Resave sections

Any help would be greatly appreciated!

### Craft CMS version

3.7.42

### PHP version

7.3.31

### Operating system and version

Linux 3.10.0-1160.45.1.el7.x86_64

### Database type and version

MYSQL 5.5.5

### Image driver and version

GD 7.3.31

### Installed plugins and versions

-

**Comment by joelstrom** [comment]
Thanks for the reply.

It is indeed only that one environment we are having trouble with, other environments such as our local ones as well as the production environment work normally for now. 

The weird thing is that we are unable to replicate the same issue on other environments, even with the same project config files and database, but at the same time it does not seem to point at a server misconfiguration as it mentions the element UID?

The environment in question has been used for a long time (years), and I don't remember stumbling on this issue before. 
A lot of things are constantly being updated on it.
Admin changes are allowed, but otherwise the setup has been done following the documentation.

I sent the files you requested to your support, any help would be greatly appreciated!

**Comment by brandonkelly** [comment]
If admin changes are allowed on that environment, how are you handling changes to its project config? It sounds to me like maybe there was a bad merge at some point, which resulted in a volume config with a UID referencing a field layout that doesn’t exist anymore.

**Comment by joelstrom** [comment]
Thanks Brandon, we will have to think through our process to avoid more of these kinds of issues in the future.

For now we managed to resolve the issue with the help of your support:

- We had a rogue field referenced in one of our entry layouts that did not exist in the database. We went into that entry layout, renamed the title, saved, renamed the title back to what it was again and saved
- One of the images in the assets table did not have a width, height, size or dateModified values so we set in a value in each of them ('10' in the first three, NOW() in the last one to get the current date)

Unsure of what caused the last issue to appear, but will try to keep that in mind in case any similar issues arise again!
